### PR TITLE
Lab 11

### PR DESCRIPTION
**Pull request by danylykkk**

### CODE REVIEW

File: P11.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import threading
+
+class RWLock:
```
Comment (polyakovartem) [inline P11.py]: it will not work like this, please use threading.Lock for write syncronization

**Review by polyakovartem — CHANGES_REQUESTED** [review]
this will not work correctly

File: P11.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    def __init__(self):
+        self.rw = 0
+        self.write_requests = 0
+        self.monitor = threading.Lock()
+        self.readers = threading.Condition(self.monitor)
+        self.writers = threading.Condition(self.monitor)
+
+    def read_lock (self):
+        self.monitor.acquire()
```
Comment (polyakovartem) [inline P11.py]: acquire should only happen if the writer thread is present. Otherwise, we have no benefits

**Review by polyakovartem — CHANGES_REQUESTED** [review]

**Comment by danylykkk** [comment]
Should I just check if self.write_requests > 0 ?

**Review by polyakovartem — CHANGES_REQUESTED** [review]
please review this example https://gist.github.com/tylerneylon/a7ff6017b7a1f9a506cf75aa23eacfd6

File: P11.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+from contextlib import contextmanager
+import threading
+
+class RWLock:
+    def __init__(self):
+        self.read_requests = 0
```
Comment (polyakovartem) [inline P11.py:6]: can't you use recursive lock instead of this variable?

**Review by polyakovartem — CHANGES_REQUESTED** [review]

**Pull request closed without merging**